packages/fpc.rb

```diff
@@ -3,56 +3,48 @@
 class Fpc < Package
   description 'Free Pascal is a 32, 64 and 16 bit professional Pascal compiler.'
   homepage 'https://www.freepascal.org/'
-  version '3.2.2'
+  version '3.2.3'
   license 'GPL-2 and LGPL-2.1-with-linking-exception'
   compatibility 'all'
-  case ARCH
-  when 'aarch64', 'armv7l'
-    source_url 'https://downloads.sourceforge.net/project/freepascal/Linux/3.2.2/fpc-3.2.2.arm-linux.tar'
-    source_sha256 '971bbcaa1934286ccdaf3998d338afc19a33235e910f8881c75579ed5d6adefd'
-  when 'i686'
-    source_url 'https://downloads.sourceforge.net/project/freepascal/Linux/3.2.2/fpc-3.2.2.i386-linux.tar'
-    source_sha256 'f62980ac0b2861221f79fdbff67836aa6912a4256d4192cfa4dfa0ac5b419958'
-  when 'x86_64'
-    source_url 'https://downloads.sourceforge.net/project/freepascal/Linux/3.2.2/fpc-3.2.2.x86_64-linux.tar'
-    source_sha256 '5adac308a5534b6a76446d8311fc340747cbb7edeaacfe6b651493ff3fe31e83'
-  end
-  binary_compression 'tar.xz'
+  source_url 'https://gitlab.com/freepascal.org/fpc/source.git'
+  git_hashtag version
+  binary_compression 'tar.zst'
 
   binary_sha256({
-    aarch64: '84701dedbee28a6db2386366318a8c2e27c389a758ea8033502d40fa2f4ec588',
-     armv7l: '84701dedbee28a6db2386366318a8c2e27c389a758ea8033502d40fa2f4ec588',
-       i686: '20062e277f6432b76115a74b6b46a38888ef696f15bbf79efce8546a1aff9a46',
-     x86_64: 'b5b9152ce23f5e7f33ab8b788ef80ca440d2d26ef300321f1e944c24051f9b45'
+    aarch64: 'ec794516f488468c8375cfc16ae2ea5e7a9cd1dc41f67f9129c6f34992bfcae9',
+     armv7l: 'ec794516f488468c8375cfc16ae2ea5e7a9cd1dc41f67f9129c6f34992bfcae9',
+       i686: 'dd54485d5db294e4cd713debc7e09b3f00150b46574dfd7bf16b21e781428729',
+     x86_64: '1264a9601eb7258c0482cf53746c0b1ae3cb06871b6d7c4ffad135d62935abeb'
   })
 
-  def self.patch
-    system "sed -i 's,PREFIX=\"\$HOME/fpc-\$VERSION\",PREFIX=#{CREW_DEST_PREFIX},' install.sh"
-    system "sed -i 's,\$PREFIX/lib,#{CREW_DEST_LIB_PREFIX},' install.sh"
+  print_source_bashrc
+
+  def self.build
+    # https://gitlab.com/freepascal.org/fpc/source/-/issues/40841
+    # fpc isn't fooled by our docker emulation, and thus disables reading from the config because it thinks we're cross-compiling.
+    # See https://gitlab.com/freepascal.org/fpc/source/-/commit/898a4debcab843b2f6f5c29b795a517485582dfb
+    # Because we don't read from the config, we don't get the correct library paths.
+    # The correct thing to do here would be to find the fault in our docker setup that reveals we're running under emulation, and fix it.
+    # In the meantime, though, just pass the correct library path manually.
+    # Chromebrew messes things up if we don't pass this in one string, adding single quotes around OPT which break things.
+    system "make OVERRIDEVERSIONCHECK=1 OPT='#{"-Fl#{CREW_LIB_PREFIX}" unless ARCH.eql?('x86_64')} -k-L#{CREW_LIB_PREFIX}' all"
```

Comment:
The nested double quotes should be escaped or removed altogether.  That might be your issue here.

Reply:
The nested double quotes work, so I'm not entirely sure what you mean?

Reply:
The system command will only run `system "make OVERRIDEVERSIONCHECK=1 OPT='#{"` and ignore the rest of the line unless you escape it.  That's what I mean.

Reply:
This does not happen, though, and I was able to build the package successfully.

Reply:
I wouldn't know how that is possible.

Reply:
If I had to guess, its probably because the quotes are inside a `#{}`, so only the result of that segment of code is then substituted in, and the result of that code does not include any double quotes.